lib/rdoc/markup/hard_break.rb

```diff
@@ -1,31 +1,34 @@
 # frozen_string_literal: true
-##
-# A hard-break in the middle of a paragraph.
 
-class RDoc::Markup::HardBreak
-
-  @instance = new
-
-  ##
-  # RDoc::Markup::HardBreak is a singleton
-
-  def self.new
-    @instance
-  end
-
-  ##
-  # Calls #accept_hard_break on +visitor+
-
-  def accept(visitor)
-    visitor.accept_hard_break self
+module RDoc
+  class Markup
+    # A hard-break in the middle of a paragraph.
+    class HardBreak < Element
+      @instance = new
+
+      # RDoc::Markup::HardBreak is a singleton
+      #: () -> HardBreak
+      def self.new
+        @instance
+      end
+
+      # Calls #accept_hard_break on +visitor+
+      # @override
+      #: (untyped) -> void
+      def accept(visitor)
+        visitor.accept_hard_break(self)
+      end
+
+      #: (top) -> bool
+      def ==(other) # :nodoc:
+        self.class === other
+      end
+
+      # @override
```

Comment:
I think this is [Sorbet-specific RBS annotation](https://sorbet.org/docs/rbs-support#method-annotations). Do you know if RBS will officially adopt it or a similar syntax?

Reply:
Yes, indeed. The RBS ecosystem doesn't have support for abstract/override yet. I assume support will be added at some point, but I'm not sure about the syntax.

For now, since we're not running a type checker on the codebase, these are more documentation than actual type annotations.

Reply:
I see. I'm a bit worried that this could confuse other maintainers/contributors that are now familiar with Ruby's typing development. Do you mind mentioning we're using Sorbet flavored RBS purely for documentation in `CONTRIBUTING.md` and `AGENTS.md`?

Reply:
Yeah, I'll open a separate PR to update it.